### PR TITLE
Migrated project details tests

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-1925

## Description
migrated playwright test frontend/src/__tests__/integration/pages/projects/ProjectDetails.spec.ts 

## How Has This Been Tested?
npm run test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      method: 'POST',
+      pathname: '/api/proxy/apis/v1beta1/pipelines',
+    },
+    buildMockPipelines(isEmpty ? [] : [mockPipelineKF({})]),
+  );
+};
+
+describe('Project Details', () => {
+  it('Empty project', () => {
+    initIntercepts({ isEmpty: true });
```
Comment (manaswinidas) [inline frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts:180]: can you put these in `beforeEach()`? both `initIntercepts` and `projectDetails.visit('test-project');`

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts (hunk 1 above)
Comment (pnaik1) [reply frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts:180]: initIntercepts has different parameters passed in every test cases

Comment (manaswinidas) [reply frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts:180]: oh yes, didn't see that, my bad.

File: frontend/src/__tests__/cypress/cypress/pages/projects.ts [hunk 2]
```diff
@@ -74,6 +74,31 @@ class ProjectDetails {
...
+
+  findComponent(componentName: string) {
+    return cy.findByTestId(componentName);
+  }
+
+  findEmptyState(componentName: string) {
+    return this.findComponent(componentName).findByTestId('empty-state-title');
+  }
+
+  findDivider() {
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/projects.ts:93]: Rename function with `should` prefix and return `this` for chaining. I don't see the return selector as having much value for chaining.

**Comment by pnaik1** [comment]
@christianvogt updated the changes

File: frontend/src/__tests__/cypress/cypress/pages/projects.ts [hunk 3]
```diff
@@ -74,6 +74,32 @@ class ProjectDetails {
...
+    return this.findComponent(componentName).findByTestId('empty-state-title');
+  }
+
+  shouldDivide() {
+    cy.findAllByTestId('details-page-section').then((sections) => {
+      cy.get('.odh-details-section--divide').then((divider) => {
+        const sectionLength = Cypress.$(sections).length;
+        expect(divider).to.have.length(sectionLength - 1);
+      });
+    });
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/projects.ts:99]: When you use a `cy.get` it searches from the page root. In this case you want to search within the scope of each section by using `find`. To do this you can use `cy.wrap` on the jquery result of the first selector to get back into the cypress chain of commands. Then proceed with finding the dividers and chaining an assertion using `should`:
```suggestion
    cy.findAllByTestId('details-page-section').then((sections) => {
      cy.wrap(sections)
        .find('.odh-details-section--divide')
        .should('have.length', sections.length - 1);
    });
```

**Pull request merged**